app/services/ProcessCsvService.scala

```diff
@@ -0,0 +1,273 @@
+    val rowStrings: Seq[String] = rowBytes.map(byteString => byteString.utf8String)
+    val parsedRow = formatDataToValidate(rowStrings, sheetInfo)
+    Try {
+      validator.validateRow(Row(0, getCells(parsedRow, 0)))
+    } match {
+      case Failure(exception) =>
+        Logger.error(s"[ProcessCsvService][processRow] Exception returned when attempting to validate row: ${exception.getMessage}")
+        Left(exception)
+      case Success(list) if list.isEmpty => Right(parsedRow)
+      case Success(_) =>
+        auditEvents.fileProcessingErrorAudit(schemeInfo, sheetName, "Failure to validate")
+        Logger.warn("[ProcessCsvService][processRow] Row validation found validation errors")
+        Left(ERSFileProcessingException(
+          s"${ErrorResponseMessages.dataParserFileInvalid}",
+          s"${ErrorResponseMessages.dataParserValidationFailure}"))
+    }
+  }
+
+  //TODO Old version - remove after successful release of large file changes
+  def sendSchemeDataCsv(ersSchemeData: SchemeData, empRef: String)(implicit hc: HeaderCarrier, request: Request[_]): Future[Option[Throwable]] = {
+    Logger.debug("[ProcessCsvService][sendSchemeDataCsv] Sheetdata sending to ers-submission " + ersSchemeData.sheetName)
+    ersConnector.sendToSubmissions(ersSchemeData, empRef).map {
+      case Right(_) =>
+        auditEvents.fileValidatorAudit(ersSchemeData.schemeInfo, ersSchemeData.sheetName)
+        None
+      case Left(ex) =>
+        auditEvents.auditRunTimeError(ex, ex.getMessage, ersSchemeData.schemeInfo, ersSchemeData.sheetName)
+        Logger.error(s"[ProcessCsvService][sendSchemeDataCsv] Exception found when sending to submissions: ${ex.getMessage}")
+        Some(ERSFileProcessingException(ex.toString, ex.getStackTrace.toString))
+    }
+  }
+
+  def sendSchemeDataCsv(ersSchemeData: SubmissionsSchemeData, empRef: String)(implicit hc: HeaderCarrier, request: Request[_]): Future[Option[Throwable]] = {
+    Logger.debug("[ProcessCsvService][sendSchemeDataCsv] Sheetdata sending to ers-submission " + ersSchemeData.sheetName)
+    ersConnector.sendToSubmissionsNew(ersSchemeData, empRef).map {
+      case Right(_) =>
+        auditEvents.fileValidatorAudit(ersSchemeData.schemeInfo, ersSchemeData.sheetName)
+        None
+      case Left(ex) =>
+        auditEvents.auditRunTimeError(ex, ex.getMessage, ersSchemeData.schemeInfo, ersSchemeData.sheetName)
+        Logger.error(s"[ProcessCsvService][sendSchemeDataCsv] Exception found when sending to submissions: ${ex.getMessage}")
+        Some(ERSFileProcessingException(ex.toString, ex.getStackTrace.toString))
+    }
+  }
+
+  def sendSchemeCsv(schemeData: SubmissionsSchemeData, empRef: String)(
+    implicit hc: HeaderCarrier, request: Request[_]) = {
+    val maxNumberOfRows: Int = appConfig.maxNumberOfRowsPerSubmission
+    val slices: Int = ValidationUtils.numberOfSlices(schemeData.numberOfRows, maxNumberOfRows)
+
+    Logger.error("number of slices is " + slices)
```

Comment:
Should this log be of error level? Also, could we give the log more context? e.g [ProcessCsvService][sendSchemeCsv]

Reply:
this log should not be there at all haha, good spot!